src/Client/SidebarComponents/Navbar.fs

```diff
@@ -1,25 +1,374 @@
 module SidebarComponents.Navbar
 
+open System
+open System.Collections.Generic
+
 open Model
 open Messages
 
 open Feliz
 open Feliz.Bulma
 
+open Components.QuickAccessButton
+open Browser.Types
+open ARCtrl
+open ARCtrl.Spreadsheet
+open Shared
+open Components.Metadata
+
+open ExcelJS.Fable
+open Excel
+open GlobalBindings
+
 type private NavbarState = {
     BurgerActive: bool
     QuickAccessActive: bool
+    SwateExcelHandleMetadataModal: bool
 } with
     static member init = {
         BurgerActive = false
         QuickAccessActive = false
+        SwateExcelHandleMetadataModal = false
     }
 
-open Components.QuickAccessButton
+type ExcelMetadataState = {
+    MetadataType: ArcFilesDiscriminate option
+    Identifier: string option
+    Assay: ArcAssay option
+    Investigation: ArcInvestigation option
+    Study: (ArcStudy * ArcAssay list) option
+    Template: Template option
+} with
+    static member init = {
+        MetadataType = None
+        Identifier = None
+        Assay = None
+        Investigation = None
+        Study = None
+        Template = None
+    }
 
+type WorkSheetNames =
```

Comment:
this does not work with backwards compatibility use arctrl function

Reply:
fixed